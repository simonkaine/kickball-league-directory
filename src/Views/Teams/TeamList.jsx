import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getTeams } from '../../Services/teams.js';
// import ball from '../../assets/ball.png';

export default function TeamList() {
    const [teams, setTeams] = useState([]);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        getTeams().then((response) => setTeams(response))

        setTimeout(() => {
            setLoading(false)
        }, 600)
    }, []);

    if (loading) return <h1 style={{height: '100vh', fontSize: '3em'}}>...Loading teams</h1>;
        // <img src={ball} style={{backgroundColor: 'white'}}></img>

    return (
        <section className='Teams'>
            <div className='teamsList'>
            <h1 style={{fontFamily: 'Century Gothic', fontSize: '4em', margin: '100px 0 0 0'}}>~ Teams ~</h1>

                <ul aria-label='teams' style={{listStyleType: 'none', height: '100vh', margin: '50px 0 0 0'}}>
                    {teams.map((team) => {
                        return (
                            <li key={team.id} style={{padding: '10px 0 10px 0'}}>
                                <Link to={`/teams/${team.id}`}>
                                    {team.name}
                                </Link>
                            </li>
                        );
                    })}
                </ul>
            </div>
        </section>
    );
}