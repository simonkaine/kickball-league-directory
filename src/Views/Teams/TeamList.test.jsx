import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import TeamList from "./TeamList";

it('Should test for a detailed Team list view', async () => {
    const { container } = render(
        <MemoryRouter  >
            <TeamList />
        </MemoryRouter>
    );

    screen.getByText('...Loading teams');
    // const labelText = await screen.findByText(/Bridge City Sneakers/i, /Rose City Bing Bongs/i);
    const labelText = await screen.findByRole('list', {name: 'teams'});
    expect(labelText).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    // const teamNames = await screen.findByText('Bridge City Sneakers', { exact: false });
    // expect(teamNames).toBeInTheDocument();
});
