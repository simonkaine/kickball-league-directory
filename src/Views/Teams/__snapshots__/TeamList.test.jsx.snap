// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test for a detailed Team list view 1`] = `
<div>
  <div>
    <a
      href="/teams/create"
    >
      Add a new team
    </a>
  </div>
  <section
    class="Teams"
  >
    <div
      class="teamsList"
    >
      <h1
        style="font-family: Century Gothic; font-size: 4em; margin: 100px 0px 0px 0px;"
      >
        ~ Teams ~
      </h1>
      <ul
        aria-label="teams"
        name="teams"
        style="list-style-type: none; height: 100vh; margin: 50px 0px 0px 0px;"
      >
        <li
          style="padding: 10px 0px 10px 0px; font-size: 2em;"
        >
          <a
            href="/teams/1"
          >
            Bridge City Sneakers
          </a>
          <a
            href="/teams/1/update"
          >
            <button>
              Edit Team
            </button>
          </a>
          <button
            aria-label="delete team Bridge City Sneakers"
            type="button"
          >
            Delete This Team
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 2em;"
        >
          <a
            href="/teams/2"
          >
            Rose City Bing Bongs
          </a>
          <a
            href="/teams/2/update"
          >
            <button>
              Edit Team
            </button>
          </a>
          <button
            aria-label="delete team Rose City Bing Bongs"
            type="button"
          >
            Delete This Team
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 2em;"
        >
          <a
            href="/teams/3"
          >
            The Dang Turkeys
          </a>
          <a
            href="/teams/3/update"
          >
            <button>
              Edit Team
            </button>
          </a>
          <button
            aria-label="delete team The Dang Turkeys"
            type="button"
          >
            Delete This Team
          </button>
        </li>
        <li
          style="padding: 10px 0px 10px 0px; font-size: 2em;"
        >
          <a
            href="/teams/4"
          >
            The Seattle Supabases
          </a>
          <a
            href="/teams/4/update"
          >
            <button>
              Edit Team
            </button>
          </a>
          <button
            aria-label="delete team The Seattle Supabases"
            type="button"
          >
            Delete This Team
          </button>
        </li>
      </ul>
    </div>
  </section>
</div>
`;
